app/models/subscription.rb

```diff
@@ -135,4 +135,20 @@ def display_name
   def invoice_name
     name.presence || plan.invoice_name
   end
+
+  # When upgrade, we want to bill one day less since date of the upgrade will be
+  # included in the first invoice for the new plan
+  def date_diff_with_timezone(from_datetime, to_datetime)
+    number_od_days = Utils::DatetimeService.date_diff_with_timezone(
+      from_datetime,
+      to_datetime,
+      customer.applicable_timezone,
+    )
+
+    return number_od_days unless terminated? && upgraded?
+
+    number_od_days = number_od_days - 1
+
+    number_od_days < 0 ? 0 : number_od_days
```

Comment:
Style/NumericPredicate: Use `number_od_days.negative?` instead of `number_od_days < 0`. (https://rubystyle.guide#predicate-methods)